applications/CompressiblePotentialFlowApplication/custom_processes/apply_far_field_process.cpp

```diff
@@ -0,0 +1,128 @@
+        const double distance_projection = inner_prod(r_coordinates, mFreeStreamVelocity);
+        if (distance_projection < min_projections[thread_id]){
+            min_projections[thread_id] = distance_projection;
+            nodes_id_list[thread_id] = it_node->Id();
+        }
+    }
+
+    // Find minimum across all threads
+    std::size_t minimum_node_thread_id = 0;
+    for (std::size_t i_thread = 0; i_thread<num_threads; i_thread++){
+        if (min_projections[i_thread] < min_projections[minimum_node_thread_id]){
+            minimum_node_thread_id = i_thread;
+        }
+    }
+
+    // Storing final result in member variable
+    mpReferenceNode = mrModelPart.pGetNode(nodes_id_list[minimum_node_thread_id]);
+}
+
+void ApplyFarFieldProcess::AssignFarFieldBoundaryConditions()
+{
+    #pragma omp parallel for
+    for (int i = 0; i < static_cast<int>(mrModelPart.Conditions().size()); i++) {
+        auto it_cond = mrModelPart.ConditionsBegin() + i;
+        auto& r_geometry = it_cond->GetGeometry();
+
+        // Computing normal
+        array_1d<double,3> aux_coordinates;
+        r_geometry.PointLocalCoordinates(aux_coordinates, r_geometry.Center());
+        const array_1d<double,3> normal = it_cond->GetGeometry().Normal(aux_coordinates);
+
+        const double velocity_projection = inner_prod(normal, mFreeStreamVelocity);
+        if (velocity_projection < 0.0) {
+            AssignDirichletFarFieldBoundaryCondition(r_geometry);
+        }
+        else {
+            AssignNeumannFarFieldBoundaryCondition(*it_cond);
+        }
+    }
+}
+
+void ApplyFarFieldProcess::AssignDirichletFarFieldBoundaryCondition(Geometry<NodeType>& rGeometry)
+{
+    // Fixing nodes in the domain that are part of the inlet, and initializing its value
+    // according to the free stream velocity.
+    for (std::size_t i_node = 0; i_node < rGeometry.size(); i_node++){
+        array_1d<double,3> relative_coordinates = rGeometry[i_node].Coordinates() - mpReferenceNode->Coordinates();
+        const double inlet_potential = inner_prod(relative_coordinates, mFreeStreamVelocity);
+        // Checking if its the primal or the adjoint case.
+        if (!rGeometry[i_node].SolutionStepsDataHas(ADJOINT_VELOCITY_POTENTIAL)) {
+            rGeometry[i_node].SetLock();
+            rGeometry[i_node].Fix(VELOCITY_POTENTIAL);
+            rGeometry[i_node].FastGetSolutionStepValue(VELOCITY_POTENTIAL) = inlet_potential + mReferencePotential;
+            rGeometry[i_node].UnSetLock();
+        }
+        else {
+            rGeometry[i_node].SetLock();
+            rGeometry[i_node].Fix(ADJOINT_VELOCITY_POTENTIAL);
+            rGeometry[i_node].FastGetSolutionStepValue(ADJOINT_VELOCITY_POTENTIAL) = 0.0;
+            rGeometry[i_node].UnSetLock();
+        }
+    }
+}
+
+void ApplyFarFieldProcess::AssignNeumannFarFieldBoundaryCondition(Condition& rCondition)
+{
+    rCondition.SetValue(FREE_STREAM_VELOCITY, mFreeStreamVelocity);
```

Comment:
a function for just one line?

Reply:
Does this affect performance?

The idea is to put a proper name to what we are doing so it's easier to read in the "main" function:

https://github.com/KratosMultiphysics/Kratos/blob/2d0cab0027da8230ac3c479ecc411e0bc8ca44aa/applications/CompressiblePotentialFlowApplication/custom_processes/apply_far_field_process.cpp#L75-L81

I could just add a comment if it affects performance.